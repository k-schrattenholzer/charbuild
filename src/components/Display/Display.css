.Display {
  display: flex;
  flex-direction: column-reverse;
  align-items: flex-start;
  width: min-content;
  background-color: var(--black-coffee); 
  color: var(--magnolia);
  padding: 10px;
  border-radius:  30px;
  box-shadow: var(--black-coffee) 3px 4px 20px;
  margin-top: 1em;
  font-size:2em;
  width: fit-content;
}