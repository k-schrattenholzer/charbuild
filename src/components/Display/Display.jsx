import React from 'react'

export default function Display({sayings, count}) {
  return (
    <div className='Display'>
      <p
      aria-label='count'>
        you've added {count} thoughts</p>

      <label
      aria-label='bucket'>
        <h3
        >the brain bucket</h3>
        {sayings.map((saying) => (
            <p key={saying}>{saying}</p>
        ))}
      </label>
      
    </div>
  )
}
