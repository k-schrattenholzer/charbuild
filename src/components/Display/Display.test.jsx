import { render, screen } from "@testing-library/react";
import Display from "./Display";

it('should display the Display component', () => {
  const testSayings = ['boy howdy', 'well hey there, cowpoke', 'proud of u']
  const {container}= render(<Display sayings={testSayings} count={3} />);

  const count = screen.getByLabelText(/count/);
  const bucket = screen.getByLabelText(/bucket/);

  expect(count).toBeInTheDocument();
  expect(bucket).toBeInTheDocument();

  expect(container).toMatchSnapshot();
});