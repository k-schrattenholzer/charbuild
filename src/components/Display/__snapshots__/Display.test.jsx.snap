// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display the Display component 1`] = `
<div>
  <div
    class="Display"
  >
    <p
      aria-label="count"
    >
      added 
      3
       thoughts
    </p>
    <label
      aria-label="bucket"
    >
      <h3>
        the brain bucket
      </h3>
      <p>
        boy howdy
      </p>
      <p>
        well hey there, cowpoke
      </p>
      <p>
        proud of u
      </p>
    </label>
  </div>
</div>
`;
